src/quoting-client.js

```diff
@@ -35,6 +36,7 @@ class QuotingClient {
       let query = {
         source_ledger: source,
         destination_ledger: destination,
+        destination_account: destinationAccount,
```

Comment:
The payments are constructed in reverse order, so that pathfinder doesn't know the intermediate (trader) source_accounts in advance of constructing their respective payments. Passing it in could allow us to set the source_account on the very first payment in the chain, but I prefer the symmetry of providing it for none of them instead.